data/globalevents/scripts/serversave.lua

```diff
@@ -1,15 +1,22 @@
 local function ServerSave()
-	if configManager.getBoolean(configKeys.SERVER_SAVE_CLEAN_MAP) then
-		cleanMap()
-	end
-
-	if configManager.getBoolean(configKeys.SERVER_SAVE_CLOSE) then
-		Game.setGameState(GAME_STATE_CLOSED)
-	end
-
 	if configManager.getBoolean(configKeys.SERVER_SAVE_SHUTDOWN) then
 		Game.setGameState(GAME_STATE_SHUTDOWN)
+	else
+		local closeAtServerSave = configManager.getBoolean(configKeys.SERVER_SAVE_CLOSE)
+		if closeAtServerSave then
+			Game.setGameState(GAME_STATE_CLOSED)
+		end
+
+		if configManager.getBoolean(configKeys.SERVER_SAVE_CLEAN_MAP) then
+			cleanMap()
+		end
+
+		if closeAtServerSave then
+			Game.setGameState(GAME_STATE_NORMAL)
+		end
 	end
+
+	saveServer()
```

Comment:
I don't think this makes sense in `GAME_STATE_SHUTDOWN` state so probably belongs to the `else` case

Reply:
I'd also move it above cleanMap. It's better to save before you start removing items bevause the server may crash when something tries to reference a removed item.

Reply:
I don't know if I understand correctly, but I believe that it is now working well.